### PR TITLE
[PostgreSQL] Add logs

### PR DESCRIPTION
**Pull request by moxarth-elastic**
## Part Of https://github.com/elastic/connectors/issues/2299

Adding more logs in PostgreSQL connector.

Log file: https://drive.google.com/file/d/12Pkq94J3ymnVGhXBtNc8S61xLqzDFCSK/view?usp=drive_link

#### Pre-Review Checklist
- [x] this PR does NOT contain credentials of any kind, such as API keys or username/passwords (double check `config.yml.example`)
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

File: connectors/sources/postgresql.py [hunk 1]
```diff
@@ -218,6 +219,12 @@ async def ping(self):
     async def get_tables_to_fetch(self, is_filtering=False):
         tables = configured_tables(self.tables)
         if is_wildcard(tables) or is_filtering:
+            msg = (
+                "Fetching all tables as the configuration field 'tables' is set to '*'"
+                if not is_filtering
+                else "Fetching all tables as the advanced sync rules are enabled."
+            )
+            self._logger.info(msg)
```
Comment (seanstory) [inline connectors/sources/postgresql.py:227]:
```suggestion
            msg = "Fetching all tables"
            self._logger.info(msg)
```

File: connectors/sources/postgresql.py [hunk 2]
```diff
@@ -266,9 +274,12 @@ async def get_table_primary_key(self, table):
             )
         ]
 
+        self._logger.debug(f"Found primary keys for '{table}' table")
+
         return primary_keys
 
     async def get_table_last_update_time(self, table):
+        self._logger.debug(f"Fetching last updated time for table: {table}")
```
Comment (seanstory) [inline connectors/sources/postgresql.py:282]: Let's log the time we fetched

File: connectors/sources/postgresql.py [hunk 3]
```diff
@@ -266,9 +274,12 @@ async def get_table_primary_key(self, table):
             )
         ]
 
+        self._logger.debug(f"Found primary keys for '{table}' table")
```
Comment (seanstory) [inline connectors/sources/postgresql.py:277]: let's log what the primary keys were

File: connectors/sources/postgresql.py [hunk 4]
```diff
@@ -300,6 +311,7 @@ async def data_streamer(
             list: It will first yield the column names, then data in each row
         """
         if query is None and row_count is not None and order_by_columns is not None:
+            self._logger.debug(f"Streaming records from database using query: {query}")
```
Comment (seanstory) [inline connectors/sources/postgresql.py:314]: This seems wrong. `query is None` was just asserted above this

File: connectors/sources/postgresql.py [hunk 5]
```diff
@@ -328,6 +340,7 @@ async def data_streamer(
                 if row_count <= offset:
                     return
         else:
+            self._logger.debug(f"Streaming records from database for table: {table}")
```
Comment (seanstory) [inline connectors/sources/postgresql.py:343]: In the sample log file, table is `None`. That seems like an error.

```
[FMWK][18:58:23][DEBUG] [Connector id: 3S2VrY4BAqKLTyj6P7_3, index name: postgresql, Sync job id: KpLkvY4BBpk9ZgBA2rJn] Streaming records from database for table: None
```

**Comment by artem-shelkovnikov** [comment]
Copying from another PR - that's applicable to all SQL connectors:

The most useful connector logs are - which queries were actually executed by the connectors. Can you add debug logs that state which exact queries were executed? A good addition to it would be to also debug log the number of records returned by the query.

File: connectors/sources/postgresql.py (hunk 4 above)
Comment (moxarth-elastic) [reply connectors/sources/postgresql.py:314]: I think the logs are swapped for if-else conditions.

**Comment by moxarth-elastic** [comment]
> which queries were actually executed by the connectors. Can you add debug logs that state which exact queries were executed?

@artem-shelkovnikov  This is already present here https://github.com/elastic/connectors/blob/43ee46ce2852c3803a85656ee37ab1f02f2f3b40/connectors/sources/postgresql.py#L348

Also, i've added a log for the number of records fetched from table/query.

**Review by artem-shelkovnikov — APPROVED** [review]

**Pull request merged**